web/packages/teleport/src/Player/Player.tsx

```diff
@@ -42,16 +42,40 @@ export function Player() {
   const { sid, clusterId } = useParams<UrlPlayerParams>();
   const { search } = useLocation();
 
+  useEffect(() => {
+    document.title = `Play ${sid} • ${clusterId}`;
+  }, [sid, clusterId]);
+
   const recordingType = getUrlParameter(
     'recordingType',
     search
   ) as RecordingType;
-  const durationMs = Number(getUrlParameter('durationMs', search));
+
+  // In order to render the progress bar, we need to know the length of the session.
+  // All in-product links to the session player should include the session duration in the URL.
+  // Some users manually build the URL based on the session ID and don't specify the session duration.
+  // For those cases, we make a separate API call to get the duration.
+  const [fetchDurationAttempt, fetchDuration] = useAsync(
+    useCallback(async () => {
+      const response = await fetchSessionDuration(clusterId, sid);
+      return response;
+    }, [clusterId, sid])
+  );
 
   const validRecordingType = validRecordingTypes.includes(recordingType);
-  const validDurationMs = Number.isInteger(durationMs) && durationMs > 0;
+  const durationMs = Number(getUrlParameter('durationMs', search));
+  const shouldFetchSessionDuration =
+    !Number.isInteger(durationMs) || durationMs <= 0;
 
-  document.title = `Play ${sid} • ${clusterId}`;
+  useEffect(() => {
+    if (shouldFetchSessionDuration) {
+      fetchDuration();
+    }
+  }, [fetchDuration, shouldFetchSessionDuration]);
+
+  const combinedAttempt = fetchDurationAttempt.status
```

Comment:
I would be more explicit here (probably I'm more used to this form): 
```suggestion
  const combinedAttempt = fetchDurationAttempt.status !== ''
```

Reply:
This doesn't work. `status` is null briefly.

Reply:
Don't you mean `attempt.data`? `attempt.status` should never be null:

<img width="397" alt="image" src="https://github.com/user-attachments/assets/7d0c5cd4-d239-42e3-b375-2e3ff2e63e47" />

Reply:
I was confused. Both my version and your suggestion are incorrect though, because `fetchDurationAttempt.status` is briefly set to the empty string before it goes into a processing state, which means for a short period of time we end up using the fake success attempt.